Jetpack App (Basics): Update magic link scheme

ashiagr opened the pull request:
SubTask #14620

This PR updates the magic link scheme for the Jetpack app. It is configured using the `build.gradle` based on the selected app.

Corresponding FluxC PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2003

To test:

- Install and launch the Jetpack app.
- Select "Login with WordPress.com".
- Enter email and click Continue.
- Click "Get a login link by email".
- Check email (it should be Jetpack-ified via diff `D61257`) and click "Log in to the app".
- Notice that the link opens the Jetpack app and login is successful.

Merge Instructions:

1. Wait for merge till corresponding FluxC PR is merged to develop.
2. Create FluxC tag and update it in the build.gradle (I can take care of it once this PR is approved).
3. Remove Not Ready for Merge.
4. Merge the PR.

## Regression Notes
1. Potential unintended areas of impact
WordPress app

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually that "Get a login link by email" uses the `wordpress` scheme and opens the WordPress app from the email.

3. What automated tests I added (or what prevented me from doing so)
Added tests to check the scheme for both WordPress and Jetpack app.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ashiagr committed:
Configure magic link scheme using gradle

ashiagr committed:
Get magic link scheme from the vm based on the selected app

ashiagr committed:
Tests: Add tests for magic link scheme

ashiagr committed:
Update fluxc version

ashiagr committed:
Tests: Remove Assertions import and use assertThat directly

ashiagr committed:
Update magic link sent label for jetpack app

ParaskP7 reviewed (approved):
👋 @ashiagr !

I have reviewed and tested this PR as per the instructions, everything works as expected, great job! 🌟 🌟 

FYI: I didn't do much of a thorough testing, but this one scenario that I managed to test, in terms of end-to-end, which was to login using a magic link, with an account that has no Jetpack sites, it seems to not be working as expected, since the `No Sites` empty screen is not the expected `Jetpack` branded, but instead it is the `WordPress` branded one.

<img src="https://user-images.githubusercontent.com/9729923/118248649-3c5cf900-b4ad-11eb-9211-0e6729ce62b7.png" width="250" height="500">

ashiagr committed:
Update fluxc tag

ashiagr commented:
Thanks, @ParaskP7! As discussed, I've added it as a #14620 subtask and will address it in a separate PR. This PR can be merged with `develop` now that I've also updated `FluxC` tag.

ParaskP7 commented:
> Thanks, @ParaskP7! As discussed, I've added it as a #14620 subtask and will address it in a separate PR. This PR can be merged with develop now that I've also updated FluxC tag.

Great @ashiagr , thanks so much for taking care of that! 🌟
